examples/java/src/test/java/org/apache/beam/examples/complete/TrafficRoutesIT.java

```diff
@@ -63,14 +69,22 @@ public void testE2ETrafficRoutes() throws Exception {
     this.options.setBigQueryDataset(this.outputDatasetId);
     this.options.setBigQueryTable(this.outputTable);
     TrafficRoutes.runTrafficRoutes(options);
-
-    QueryResponse response =
-        this.bqClient.queryWithRetries(
-            String.format(
-                "SELECT count(*) as total FROM [%s:%s.%s]",
-                this.projectId, this.outputDatasetId, this.outputTable),
-            this.projectId);
-    String res = response.getRows().get(0).getF().get(0).getV().toString();
+    FluentBackoff backoffFactory =
+        FluentBackoff.DEFAULT.withMaxRetries(4).withInitialBackoff(Duration.standardSeconds(1L));
+    Sleeper sleeper = Sleeper.DEFAULT;
+    BackOff backoff = BackOffAdapter.toGcpBackOff(backoffFactory.backoff());
+    String res = "empty_result";
+    do {
+      QueryResponse response =
+          this.bqClient.queryWithRetries(
+              String.format(
+                  "SELECT count(*) as total FROM [%s:%s.%s]",
+                  this.projectId, this.outputDatasetId, this.outputTable),
+              this.projectId);
+      if (response.getRows() != null && response.getRows().get(0).getF() != null) {
```

Comment:
1. Don't we need null check at every location?
2. Can we compare TableRow (response.getRows().get(0)) object to a dummy TableRow(new TableRow().set....) object with the expected values?

Reply:
1. If every location means not only the first row, then the result table is expected to have only one row: select count.
2. I don't think TableRow is comparable directly but it can be serialized to string, which can serve as a matcher.